adapter/internal/oasparser/envoyconf/envoyconf_internal_test.go

```diff
@@ -396,6 +396,36 @@ func TestGenerateRegex(t *testing.T) {
 			message:       "when the resource ends with *, trailing characters substitution passes",
 			isMatched:     false,
 		},
+		{
+			inputpath:     "/v2.0.0/pet/{petId}/images/*",
+			userInputPath: "/v2x0x0/pet/123/images/123-foo.png",
+			message:       "when the version is semver, the dot should not matched with any character",
```

Comment:
This test case is applicable for all cases. Eg: `2.0.0` and `v2.0`

Shall we Generalize the message? Or we can add 2 separate entries to test semver enabled ones and the older APIs.

Reply:
Done with https://github.com/wso2/product-microgateway/pull/3429/commits/eec8652a242a83b98a8caed43825bcb845735744